core/vm/contracts_suave.go

```diff
@@ -133,3 +136,42 @@ type suaveRuntime struct {
 }
 
 var _ SuaveRuntime = &suaveRuntime{}
+
+func (s *suaveRuntime) doHTTPRequest(request types.HttpRequest) ([]byte, error) {
+	if request.Method != "GET" && request.Method != "POST" {
+		return nil, fmt.Errorf("only GET and POST methods are supported")
+	}
+	if request.Url == "" {
```

Comment:
I'd be a bit worried about allowing connecting to arbitrary urls

Reply:
do you have some example targets for the requests in mind? anything that needs authentication, yiou'll have to at least set the request headers. I think this "http" library also allows https urls and applies a default certificate authority policy right?

Reply:
> I'd be a bit worried about allowing connecting to arbitrary URLs

Note that there is already another precompile which allows calling arbitrary URLs [here](https://github.com/flashbots/suave-geth/blob/main/suave/gen/suave_spec.yaml#L156).

We have not discussed yet the security model of making external requests. I was thinking of that to be a follow-up conversation since it is going to take several iterations to dial down. My initial take would be to have a flag in the Kettle to whitelist a list of domains. How does that sound?

> do you have some example targets for the requests in mind?

[This](https://github.com/flashbots/suave-geth/blob/feature/suave-std/suave/sol/suave-std/suavex.sol#L33) is an example. It makes a JSON-RPC Post request to simulate a transaction in the domain-specific node.

> I think this "http" library also allows https urls and applies a default certificate authority policy right?

Yes, it uses the Go standard "HTTP" library so it uses default certificate authorities.

Reply:
I guess you are correct